bindings/rust/s2n-tls/src/config.rs

```diff
@@ -595,6 +632,70 @@ impl Builder {
         Ok(self)
     }
 
+    /// Enable negotiating session tickets in a TLS connection
+    pub fn enable_session_tickets(&mut self, enable: bool) -> Result<&mut Self, Error> {
+        unsafe {
+            s2n_config_set_session_tickets_onoff(self.as_mut_ptr(), enable.into()).into_result()
+        }?;
+        Ok(self)
+    }
+
+    /// Adds a key which will be used to encrypt and decrypt session tickets. The intro_time parameter is time since
+    /// the Unix epoch (Midnight, January 1st, 1970). If this is 0, then into_time is set to now.
+    pub fn add_session_ticket_key(
+        &mut self,
+        key_name: &[u8],
+        key: &[u8],
+        intro_time: Duration,
+    ) -> Result<&mut Self, Error> {
+        let key_name_len: u32 = key_name
+            .len()
+            .try_into()
+            .map_err(|_| Error::INVALID_INPUT)?;
+        let key_len: u32 = key.len().try_into().map_err(|_| Error::INVALID_INPUT)?;
+        unsafe {
+            s2n_config_add_ticket_crypto_key(
+                self.as_mut_ptr(),
+                key_name.as_ptr(),
+                key_name_len,
+                // s2n-tls doesn't mutate key, it's just mut for easier use with stuffers and blobs
+                key.as_ptr() as *mut u8,
+                key_len,
+                intro_time.as_secs(),
+            )
+            .into_result()
+        }?;
+        Ok(self)
+    }
+
+    // Sets how long a session ticket key will be able to be used for both encryption
+    // and decryption of tickets
+    pub fn set_ticket_key_encrypt_decrypt_lifetime(
```

Comment:
nit: Should we document that Duration will be rounded down to the nearest whole second? So if someone tries to add a Duration of 0.99 seconds, they will end up with a lifetime of 0 seconds?

On the other hand, I can't imagine why anyone would ever intentionally use the API with a ticket lifetime less than a second?

Reply:
Not super worried about this, session resumption is essentially impossible if the ticket lifetime is less than a second anyways :)